src/psyclone/domain/lfric/lfric_symbol_table.py

```diff
@@ -168,7 +168,7 @@ def find_or_create_array(self, array_name, num_dimensions, intrinsic_type,
             # pylint: disable=raise-missing-from
             # Create a DataSymbol for this kernel argument.
             array_type = ArrayType(datatype,
-                                   [ArrayType.Extent.ATTRIBUTE]*num_dimensions)
+                                   [ArrayType.Extent.DEFERRED]*num_dimensions)
```

Comment:
Please extend L134 to say that any new symbol is given DEFERRED extent which, in Fortran, represents an allocatable array.

Reply:
Done.